core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -732,10 +756,27 @@ public PersistenceState getPersistenceState(long fileId) throws FileDoesNotExist
             && ((InodeDirectory) inode).isDirectChildrenLoaded()) {
           loadMetadataOptions.setLoadDirectChildren(false);
         }
+      } else {
+        if (listStatusOptions.getLoadMetadataType() == LoadMetadataType.Never || (
```

Comment:
optional: this is duplicated in the code and could be factored out into a method